spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -183,6 +187,8 @@ private Expression filterExpression() {
 
   private StructType lazyType() {
     if (type == null) {
+      Preconditions.checkArgument(readTimestampWithoutZone || !SparkUtil.hasTimestampWithoutZone(lazySchema()),
+              SparkUtil.TIMESTAMP_WITHOUT_TIMEZONE_ERROR);
```

Comment:
Would it make sense to add this check into `SparkSchemaUtil.convert`?

Reply:
I think no, because we need this check only in few places, while SparkSchemaUtil.convert used in a lot of places

Reply:
For reference, this is something just to consider and not a strong mandate or requirement to merge.